### DISCUSSION TITLE
Sming-Graphics Status

### DISCUSSION THREAD

**Discussion by callimero** [SmingHub/Sming#2797] [General]
Hi,

I really like the idea of Sming-Graphics. It works for me ok with the virtual screen. So then I learned that only the ILI9341 is supported so far. I checked the GitHub and it seems it is a bit quiet there since years.

So how is the status/roadmap of Sming-Graphics?
I managed to grab a IL9341 out of a box with abandoned projects, worth trying? 
I still have no clue how to compile for a physical tft, did not find a sample or where I need to set the right GPIOs...

All the best,
Carsten

**Comment by mikee47** [comment]
Hello, glad you like the library! It's mainly just me working on it, unlikely many people are using it so if you need something with a broader user base may not be for you!

The documentation is here https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/Graphics/index.html and there's some [references](https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/Graphics/index.html#references) to other libraries although as yet not ported to Sming. These all operate in a more bursty fashion, i.e. 'fast as possible' rather than '(slow) and steady gets it done'.

At the [bottom](https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/Graphics/index.html#used-by) of the page is a list of samples you can try. These are to be found in `Sming/Libraries/Graphics/samples`. Main header files are to be found in `src/include/Graphics`. The samples all use a common configuration file in 'SampleConfig.h` so you can edit that to suit your particular hardware. Under 'Display' you can find the displays the library's been used with.

**Comment by mikee47** [comment]
> So how is the status/roadmap of Sming-Graphics?

I use it with a raspberry pi pico-w and ILI9341 display where it's been running happily for a couple of years. There's an experimental branch `feature/graphical-editor` which I hope to merge sometime soon. Go to the `Graphic_Editor` sample and type `make SMING_SOC=host graphic-editor` to have a go.

**Reply by callimero** [reply]
I will try that! :) Sounds fun.

**Reply by callimero** [reply]
What is GED? I only found some references which were obviously not what I looked for :)

Carsten

**Reply by mikee47** [reply]
GED === Graphical EDitor

**Reply by callimero** [reply]
Yea, but Googling did not bring up a thing that looked like it.

C!

**Reply by mikee47** [reply]
```
cd $SMING_HOME/Libraries/Graphics
git checkout feature/graphical-editor
cd samples/Graphic_Editor
make graphic-editor
```

**Reply by slaff** [reply]
You can add also this command as the first one in the list above:
```
make -C $SMING_HOME fetch Graphics
```

**Reply by anakod** [reply]
Has anyone tried to run LVGL on the Sming Framework?

**Reply by slaff** [reply]
> Has anyone tried to run LVGL on the Sming Framework?

@anakod I have tried it some time ago and it was working nicely. But it had a kind-of-big incompatibility with Sming: it was not possible to make the main loop run async. Which had the potential of segfaulting the app if the screen, for example, was not fast enough to update. I have to check the latest version and see if this is still the case.
